shimCPURepro/backend_mismatch_reproducer.cpp

```diff
@@ -0,0 +1,257 @@
+#include <hip/hip_runtime.h>
+#include <hip/hip_interop.h>
+#include <iostream>
+#include <vector>
+#include <cmath>
+#include <cstring>
+
+#define HIP_CHECK(status) \
+    if (status != hipSuccess) { \
+        std::cerr << "HIP error: " << hipGetErrorString(status) << " at line " << __LINE__ << std::endl; \
+        exit(1); \
+    }
+
+// For hipGetBackendNativeHandles which returns int, not hipError_t
+#define INTEROP_CHECK(status) \
+    if (status != 0) { \
+        std::cerr << "Interop error: " << status << " at line " << __LINE__ << std::endl; \
+        exit(1); \
+    }
+
+// Handle indices for debugging
+const char* HandleNames[] = {
+    "BACKEND_NAME",
+    "PLATFORM_DRIVER", 
+    "DEVICE", 
+    "CONTEXT", 
+    "QUEUE", 
+    "COMMAND_LIST"
+};
+
+// More detailed handle printing with names
+void print_handles(const char* prefix, const std::vector<uintptr_t>& handles) {
+    std::cout << "\n=== " << prefix << " (" << handles.size() << " handles) ===" << std::endl;
+    
+    for (size_t i = 0; i < handles.size(); i++) {
+        const char* name = (i < sizeof(HandleNames)/sizeof(HandleNames[0])) ? 
+                       HandleNames[i] : "UNKNOWN";
+        std::cout << "Handle[" << i << "] " << name << ": 0x" << std::hex 
+                 << handles[i] << std::dec << std::endl;
+    }
+    std::cout << std::endl;
+}
+
+// Compare two sets of handles and print differences
+void compare_handles(const char* name1, const std::vector<uintptr_t>& handles1, 
+                     const char* name2, const std::vector<uintptr_t>& handles2) {
+    std::cout << "\n=== Comparing " << name1 << " vs " << name2 << " ===" << std::endl;
+    
+    size_t minSize = std::min(handles1.size(), handles2.size());
+    size_t maxSize = std::max(handles1.size(), handles2.size());
+    
+    bool hasDifference = false;
+    
+    // First check if sizes are different
+    if (handles1.size() != handles2.size()) {
+        std::cout << "DIFFERENCE: Handle count mismatch - " << name1 << ": " 
+                 << handles1.size() << ", " << name2 << ": " << handles2.size() << std::endl;
+        hasDifference = true;
+    }
+    
+    // Compare handles that both arrays have
+    for (size_t i = 0; i < minSize; i++) {
+        const char* name = (i < sizeof(HandleNames)/sizeof(HandleNames[0])) ? 
```

Comment:
"sizeof(HandleNames)/sizeof(HandleNames[0])" is a fixed value why do we need to calculate in every iteration of the loop?